app/controllers/users/approvals_controller.rb

```diff
@@ -10,45 +10,36 @@ def new
   end
 
   def create
-    result = CreateApproval.call(
+    result = Users::Approvals::CreateUserApproval.call(
       current_user: current_user,
       approvable: approval_params[:approvable]
     )
     approvals_presenter_and_gon("User") if result.success?
     redirect_to(result.path, result.message)
   end
 
-  def apps
-    approvals_presenter_and_gon('Developer')
-    render 'approvals'
+  def index
+    approvals_presenter_and_gon(params[:approvable_type])
+    render "approvals"
   end
 
-  def friends
-    approvals_presenter_and_gon('User')
-    render 'approvals'
-  end
-
-  def approve
-    approval = Approval.find_by(id: params[:id], user: current_user)
-    approvable_type = approval.approvable_type
-    approvable = approval.approvable
-    Approval.accept(current_user, approvable, approvable_type)
-    approvals_presenter_and_gon(approvable_type)
-    approvable.notify_if_subscribed('new_approval', approval_zapier_data(approval)) if approvable_type == 'Developer'
+  def update
+    result = Users::Approvals::UpdateApproval.call(
+      current_user: current_user,
+      params: params
+    )
+    approvals_presenter_and_gon(result.approvable_type)
+    return unless result.approvable_type == "Developer"
+    result.approvable.notify_if_subscribed("new_approval", approval_zapier_data(result.approval))
```

Comment:
approval_zapier_data is a shared controller method which can't be accessed from an interactor, perhaps I could move this method to some shared interactor file? Not sure where to put methods shared between interactors.